### ISSUE TITLE
kinesis-firehose recreates during import

### ISSUE THREAD

**Issue by Kydybets** [terraform-providers/terraform-provider-aws#13530]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

terraform -v
Terraform v0.12.24
aws.provider v2.61.0

### Affected Resource(s)

arn:aws:firehose:***

### Terraform Configuration Files

Once we import new `firehose` and try to plan, it tries to recreate the resource

```hcl
resource "aws_kinesis_firehose_delivery_stream" "extended_s3_stream" {
  name        = var.kinesis_firehose_stream_name
  destination = var.destination_type

  kinesis_source_configuration {
    kinesis_stream_arn = var.kinesis_source_stream_arn
    role_arn = var.iam_firehose_role_arn
  }

  extended_s3_configuration {
    role_arn   = var.iam_firehose_role_arn
    bucket_arn = var.s3_destination_bucket_arn
    buffer_interval    = var.extended_s3_configuration_buffer_interval
    buffer_size        = var.extended_s3_configuration_buffer_size
    compression_format = var.extended_s3_configuration_compression_format
    s3_backup_mode     = var.extended_s3_configuration_s3_backup_mode
    prefix             = var.extended_s3_configuration_prefix
    error_output_prefix = var.extended_s3_configuration_error_output_prefix
    kms_key_arn        = var.extended_s3_configuration_kms_key_arn


    cloudwatch_logging_options {
      enabled         = var.cloudwatch_logging_options_enabled
      log_group_name  = var.cloudwatch_logging_options_log_group_name
      log_stream_name = var.cloudwatch_logging_options_log_stream_name
    }

    processing_configuration {
      enabled = var.processing_configuration_enabled
    }
  }

  tags = {
    Environment = var.environment
    Managed-by = "Terraform"
    Owner = var.owner
  }

}
```

### Expected Behavior

Should import `kinesis-firehose` without any recreation

### Actual Behavior
```
-/+ resource "aws_kinesis_firehose_delivery_stream" "extended_s3_stream" {
      ~ arn            = "<arn>" -> (known after apply)
        destination    = "extended_s3"
      ~ destination_id = "destinationId-000000000001" -> (known after apply)
      ~ id             = "<arn>" -> (known after apply)
        name           = "dev-pos-orders-new-model-deliver-s3-stream"
        tags           = {
            "Environment" = "dev"
            "Managed-by"  = "Terraform"
            "Owner"       = "test"
        }
      ~ version_id     = "7" -> (known after apply)
      ~ extended_s3_configuration {
            bucket_arn          = "<s3-arn>"
            buffer_interval     = 60
            buffer_size         = 100
            compression_format  = "GZIP"
            error_output_prefix = "error_orders/modified_date=!{timestamp:yyyy-MM-dd}/!{firehose:error-output-type}"
            prefix              = "orders/modified_date=!{timestamp:yyyy-MM-dd}/"
            role_arn            = "<role_arn>"
            s3_backup_mode      = "Disabled"
            cloudwatch_logging_options {
                enabled         = true
                log_group_name  = "<lof>"
                log_stream_name = "S3Delivery"
            }
            processing_configuration {
                enabled = false
            }
        }
      + kinesis_source_configuration { # forces replacement
          + kinesis_stream_arn = "<kinesis-arn>" # forces replacement
          + role_arn           = "<role_arn>" # forces replacement
        }
      - server_side_encryption {
          - enabled = false -> null
        }
    }
<firehose_module_name>: Destroying... [id=<arn>]
<firehose_module_name>: Still destroying... [id=arn, 10s elapsed]
<firehose_module_name>: Destruction complete after 11s
<firehose_module_name>: Creating...
<firehose_module_name>: Still creating... [10s elapsed]
<firehose_module_name>: Creation complete after 12s [id=arn]
<firehose_module_name>
The plan was created with the -target option in effect, so some changes
requested in the configuration may have been ignored and the output values may
not be fully updated. Run the following command to verify that no other
changes are pending:
    terraform plan
Note that the -target option is not suitable for routine use, and is provided
only for exceptional situations such as recovering from errors or mistakes, or
when Terraform specifically suggests to use it as part of an error message.
Apply complete! Resources: 1 added, 0 changed, 1 destroyed.
Releasing state lock. This may take a few moments...
```
### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by ewbankkit** [comment]
@Kydybets Thanks for raising this issue.
It looks like `kinesis_source_configuration` isn't being read back from the AWS API during resource import, hence it isn't set in the state and so a subsequent `terraform plan` shows the diff.
Because changes to the `kinesis_source_configuration` attribute force the resource to be recreated (`ForceNew: true`), you get the behavior you experienced.

**Comment by Kydybets** [comment]
@ewbankkit  i am not sure that i understood you right.
but once import resource, i try to plan and see this diff, and `ForceNew: true`
for test purposes, i try to do apply and see what will be changed in the `AWS` side, and resource was fully recreated.
This behavior is ok for `development`,but not for `production`

What you can suggest in this particular case?

**Comment by ewbankkit** [comment]
@Kydybets It seems to be a bug in the code.
I can reproduce with a slight modification to one of the existing acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSKinesisFirehoseDeliveryStream_s3KinesisStreamSource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSKinesisFirehoseDeliveryStream_s3KinesisStreamSource -timeout 120m
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_s3KinesisStreamSource
=== PAUSE TestAccAWSKinesisFirehoseDeliveryStream_s3KinesisStreamSource
=== CONT  TestAccAWSKinesisFirehoseDeliveryStream_s3KinesisStreamSource
--- FAIL: TestAccAWSKinesisFirehoseDeliveryStream_s3KinesisStreamSource (113.00s)
    testing.go:684: Step 1 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) (len=15) {
         (string) (len=11) "destination": (string) (len=11) "extended_s3",
         (string) (len=27) "extended_s3_configuration.#": (string) (len=1) "1",
         (string) (len=38) "extended_s3_configuration.0.bucket_arn": (string) (len=47) "arn:aws:s3:::tf-test-bucket-8737727327088504269",
         (string) (len=43) "extended_s3_configuration.0.buffer_interval": (string) (len=3) "300",
         (string) (len=39) "extended_s3_configuration.0.buffer_size": (string) (len=1) "5",
         (string) (len=56) "extended_s3_configuration.0.cloudwatch_logging_options.#": (string) (len=1) "1",
         (string) (len=64) "extended_s3_configuration.0.cloudwatch_logging_options.0.enabled": (string) (len=5) "false",
         (string) (len=71) "extended_s3_configuration.0.cloudwatch_logging_options.0.log_group_name": (string) "",
         (string) (len=72) "extended_s3_configuration.0.cloudwatch_logging_options.0.log_stream_name": (string) "",
         (string) (len=46) "extended_s3_configuration.0.compression_format": (string) (len=12) "UNCOMPRESSED",
         (string) (len=47) "extended_s3_configuration.0.error_output_prefix": (string) "",
         (string) (len=39) "extended_s3_configuration.0.kms_key_arn": (string) "",
         (string) (len=34) "extended_s3_configuration.0.prefix": (string) "",
         (string) (len=36) "extended_s3_configuration.0.role_arn": (string) (len=84) "arn:aws:iam::123456789012:role/tf_acctest_firehose_delivery_role_8737727327088504269",
         (string) (len=42) "extended_s3_configuration.0.s3_backup_mode": (string) (len=8) "Disabled"
        }
        
        
        (map[string]string) (len=16) {
         (string) (len=11) "destination": (string) (len=2) "s3",
         (string) (len=30) "kinesis_source_configuration.#": (string) (len=1) "1",
         (string) (len=49) "kinesis_source_configuration.0.kinesis_stream_arn": (string) (len=107) "arn:aws:kinesis:us-west-2:123456789012:stream/terraform-kinesis-source-stream-basictest-8737727327088504269",
         (string) (len=39) "kinesis_source_configuration.0.role_arn": (string) (len=81) "arn:aws:iam::123456789012:role/tf_acctest_kinesis_source_role_8737727327088504269",
         (string) (len=18) "s3_configuration.#": (string) (len=1) "1",
         (string) (len=29) "s3_configuration.0.bucket_arn": (string) (len=47) "arn:aws:s3:::tf-test-bucket-8737727327088504269",
         (string) (len=34) "s3_configuration.0.buffer_interval": (string) (len=3) "300",
         (string) (len=30) "s3_configuration.0.buffer_size": (string) (len=1) "5",
         (string) (len=47) "s3_configuration.0.cloudwatch_logging_options.#": (string) (len=1) "1",
         (string) (len=55) "s3_configuration.0.cloudwatch_logging_options.0.enabled": (string) (len=5) "false",
         (string) (len=62) "s3_configuration.0.cloudwatch_logging_options.0.log_group_name": (string) "",
         (string) (len=63) "s3_configuration.0.cloudwatch_logging_options.0.log_stream_name": (string) "",
         (string) (len=37) "s3_configuration.0.compression_format": (string) (len=12) "UNCOMPRESSED",
         (string) (len=30) "s3_configuration.0.kms_key_arn": (string) "",
         (string) (len=25) "s3_configuration.0.prefix": (string) "",
         (string) (len=27) "s3_configuration.0.role_arn": (string) (len=84) "arn:aws:iam::123456789012:role/tf_acctest_firehose_delivery_role_8737727327088504269"
        }
        
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	113.084s
FAIL
GNUmakefile:26: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

**Comment by Kydybets** [comment]
@ewbankkit 
is there any chance that this bug could be resolved?

**Comment by ewbankkit** [comment]
I have opened #13536 to fix this.

**Comment by Kydybets** [comment]
@ewbankkit thank you.
When it could be merged to get this fix?

**Comment by anGie44** [comment]
Thanks @Kydybets and @ewbankkit for raising the original issue and submitting the fix 🙌 
The fix for this has been merged and will release with version `2.65.0` of the Terraform AWS Provider, expected in today's release.
